### PR TITLE
add group listing  and group device listing fixes

### PR DESCRIPTION
**Pull request by sameeragunarathne**

### CODE REVIEW

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/api/DeviceManagementService.java [hunk 1]
```diff
@@ -129,6 +129,12 @@ Response getDevices(
             @Size(max = 45)
             String status,
             @ApiParam(
+                    name = "groupId",
+                    value = "The group name of the device included",
```
Comment (charithag) [inline components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/api/DeviceManagementService.java]: value = "Id of the group which device belongs"

File: components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/modules/business-controllers/group.js [hunk 2]
```diff
@@ -53,11 +53,11 @@ var groupModule = {};
         );
     };
 
-    groupModule.getGroupDeviceCount = function (groupName, owner) {
-        endPoint = deviceServiceEndpoint + "/owner/" + owner + "/name/" + groupName + "/devices/count";
+    groupModule.getGroupDeviceCount = function (groupId, owner) {
```
Comment (charithag) [inline components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/modules/business-controllers/group.js]: IMO, we can remove owner attribute from here

File: components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/pages/cdmf.page.groups/public/js/listing.js [hunk 3]
```diff
@@ -318,15 +351,14 @@ function attachEvents() {
      * on Group Management page in WSO2 IoT Server Console.
      */
     $("a.remove-group-link").click(function () {
-        var groupName = $(this).data("group-name");
+        var groupId = $(this).data("group-id");
         var groupOwner = $(this).data("group-owner");
```
Comment (charithag) [inline components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/pages/cdmf.page.groups/public/js/listing.js]: IMO, we can remove group owner field from here

File: components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/pages/cdmf.page.devices/devices.js [hunk 4]
```diff
@@ -23,6 +23,7 @@ function onRequest(context) {
 
     var groupName = request.getParameter("groupName");
     var groupOwner = request.getParameter("groupOwner");
```
Comment (charithag) [inline components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/pages/cdmf.page.devices/devices.js]: IMO, we don't need to have groupName and groupOwner in here as they can also remove from the group module. WDYT?

**Review by charithag — CHANGES_REQUESTED** [review]
@sameeragunarathne could you please review the requested changes?

File: components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/pages/cdmf.page.devices/devices.js (hunk 4 above)
Comment (sameeragunarathne) [reply components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/pages/cdmf.page.devices/devices.js]: groupName is used in the same js as to change the title of the page. Other than that it is not used for any purpose. 
if (groupName) {
        title = groupName + " " + title;
        viewModel.groupName = groupName;
    }

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/api/DeviceManagementService.java (hunk 1 above)
Comment (sameeragunarathne) [reply components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/api/DeviceManagementService.java]: fixed.

File: components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/modules/business-controllers/group.js (hunk 2 above)
Comment (sameeragunarathne) [reply components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/modules/business-controllers/group.js]: fixed.

File: components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/pages/cdmf.page.groups/public/js/listing.js (hunk 3 above)
Comment (sameeragunarathne) [reply components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/pages/cdmf.page.groups/public/js/listing.js]: fixed.

**Pull request merged**